File: 0330-patching-array/greedy/greedy.cpp
#include <vector>

using namespace std;

class Solution {
public:
  int minPatches(vector<int> &nums, int n) {
    long curSum = 0;
    int idx = 0, m = nums.size(), res = 0;
    while (curSum < n)
      if (idx < m && nums[idx] <= curSum + 1) {
        curSum += nums[idx];
        idx++;
      } else {
        curSum += curSum + 1;
        res++;
      }
    return res;
  }
};

File: 0330-patching-array/greedy/greedy.py
from typing import List

class Solution:
  def minPatches(self, nums: List[int], n: int) -> int:
    curSum = res = idx = 0
    m = len(nums)
    while curSum < n:
      if idx < m and nums[idx] <= curSum + 1:
        curSum += nums[idx]
        idx += 1
      else:
        curSum += curSum + 1
        res += 1
    return res
